src/main/java/net/spy/memcached/MemcachedConnection.java

```diff
@@ -397,15 +416,8 @@ public void notifyUpdate(ClusterConfiguration clusterConfiguration){
     
     Selector s = selector.wakeup();
     assert s == selector : "Wakeup returned the wrong selector.";
-    
-    conditionLock.lock();
-    try{
-      try {
-        nodeUpdateCondition.await(50L, TimeUnit.SECONDS);
-      } catch (InterruptedException e) { }
-    }finally{
-      conditionLock.unlock();
-    }
+
+    waitForNodeUpdateCondition();
```

Comment:
I would not do this as part of `notifyUpdate()`. I would separate this logic out to a separate function like you did yesterday.

Reply:
Separate the logic as discussed offline